packages/client/src/runtime/core/types/exported/Extensions.ts

```diff
@@ -216,38 +213,34 @@ export type DynamicModelExtensionFnResult<
   A,
   P extends PropertyKey,
   Null,
-  ClientOptions
 > = P extends FluentOperation
-  ? & DynamicModelExtensionFluentApi<TypeMap, M, P, Null, ClientOptions>
-    & PrismaPromise<DynamicModelExtensionFnResultBase<TypeMap, M, A, P, ClientOptions> | Null>
-  : PrismaPromise<DynamicModelExtensionFnResultBase<TypeMap, M, A, P, ClientOptions>>
+  ? & DynamicModelExtensionFluentApi<TypeMap, M, P, Null>
+    & PrismaPromise<DynamicModelExtensionFnResultBase<TypeMap, M, A, P> | Null>
+  : PrismaPromise<DynamicModelExtensionFnResultBase<TypeMap, M, A, P>>
 
 export type DynamicModelExtensionFnResultBase<
   TypeMap extends TypeMapDef,
   M extends PropertyKey,
   A,
   P extends PropertyKey,
-  ClientOptions,
-> = GetOperationResult<TypeMap['model'][M]['payload'], A, P & Operation, ClientOptions>
+> = GetOperationResult<TypeMap['model'][M]['payload'], A, P & Operation, TypeMap['globalOmitOptions']>
```

Comment:
ℹ️  This is the place where we can now take the "shortcut" to get the global omit config from the TypeMap instead of passed client options.